x-pack/plugins/ml/common/types/data_frame_analytics.ts

```diff
@@ -79,3 +80,5 @@ export interface DataFrameAnalyticsConfig {
   version: string;
   allow_lazy_start?: boolean;
 }
+
+export type DataFrameAnalyticsType = typeof ANALYSIS_CONFIG_TYPE[keyof typeof ANALYSIS_CONFIG_TYPE];
```

Comment:
I don't think we use the `Type` postfix in other places so suggest to just name this `DataFrameAnalytics`.